openeogeotrellis/load_stac.py

```diff
@@ -723,6 +710,120 @@ def intersects_temporally(interval) -> bool:
     return GeopysparkDataCube(pyramid=gps.Pyramid(levels), metadata=metadata)
 
 
+class _TemporalExtent:
+    """
+    Helper to represent a temporal extent with a from_date and to_date
+    and calculate intersection with STAC entities
+    (nominal datetime or start_datetime+end_datetime).
+    """
+
+    # TODO: move this to a more generic location for better reuse
+    __slots__ = ("from_date", "to_date")
+
+    def __init__(
+        self,
+        from_date: Union[str, datetime.datetime, datetime.date, None],
+        to_date: Union[str, datetime.datetime, datetime.date, None],
+    ):
+        self.from_date: Union[datetime.datetime, None] = self._to_datetime_utc_unless_none(from_date)
+        self.to_date: Union[datetime.datetime, None] = self._to_datetime_utc_unless_none(to_date)
+
+    @staticmethod
+    def _to_datetime_utc(d: Union[str, datetime.datetime, datetime.date]) -> datetime.datetime:
```

Comment:
move this helper to a more general place to make `_TemporalExtent` a bit smaller